extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -198,13 +198,16 @@ export abstract class AzureAuth implements vscode.Disposable {
 		return account;
 	}
 
-	public async getAccountSecurityTokenAdal(account: AzureAccount, tenantId: string, azureResource: azdata.AzureResource): Promise<Token | undefined> {
+	public async getAccountSecurityTokenAdal(account: AzureAccount, tenantId: string, azureResource: azdata.AzureResource | string): Promise<Token | undefined> {
 		if (account.isStale === true) {
 			Logger.error('Account was stale. No tokens being fetched.');
 			return undefined;
 		}
 
-		const resource = this.resources.find(s => s.azureResourceId === azureResource);
+		const resource = (typeof azureResource === 'string')
```

Comment:
Make sure the expected values of azureResource are documented so callers know what the string is expected to be (same for other functions that you've changed this for)